### PR TITLE
New sign up: Login Functionality

### PR DESCRIPTION
**Pull request by Carbinski**
Added login functionality.

Compared user info and password info to data base. If not found or not matching then responds with error signing in. Otherwise responds with secret token

### CODE REVIEW

**Comment by Carbinski** [comment]
Also I made some changes so it works on my local mongo db rather than the one you set up

File: index.js [hunk 1]
```diff
@@ -18,6 +53,58 @@ app.get('/hello/:name', (req, res) => {
...
+app.post('/api/login', async (req, res) => {
+  const user = {
+    username: req.body.username,
+    password: req.body.password
+  };
+
+  let userID;
+
+  try {
+    const object = await userCollection.find({
```
Comment (reiden-webber) [inline index.js]:
```suggestion
    const object = await userCollection.find(user)
```

File: index.js [hunk 2]
```diff
@@ -18,6 +53,58 @@ app.get('/hello/:name', (req, res) => {
...
+});
+
+// API to look for user data in the database
+app.post('/api/login', async (req, res) => {
+  const user = {
+    username: req.body.username,
+    password: req.body.password
+  };
+
+  let userID;
```
Comment (reiden-webber) [inline index.js:89]: remove unused line

File: index.js [hunk 3]
```diff
@@ -18,6 +53,58 @@ app.get('/hello/:name', (req, res) => {
...
+
+  try {
+    const object = await userCollection.find({
+      username: user.username,
+      password: user.password
+      });
+    console.log(object);
```
Comment (reiden-webber) [inline index.js:96]: remove debug statement

File: index.js [hunk 4]
```diff
@@ -18,6 +53,58 @@ app.get('/hello/:name', (req, res) => {
...
+    if (object.length > 0) {
+      res.status(202).send({token: TOKEN_SECRET});
```
Comment (reiden-webber) [inline index.js:98]: Send the generated token instead of TOKEN_SECRET

File: index.js [hunk 5]
```diff
@@ -18,6 +53,58 @@ app.get('/hello/:name', (req, res) => {
...
+
+  let userID;
+
+  try {
+    const object = await userCollection.find({
+      username: user.username,
+      password: user.password
+      });
+    console.log(object);
+    if (object.length > 0) {
```
Comment (reiden-webber) [inline index.js:97]: Add a comment here what does object length look like and why if it is greater than zero then it exists

File: index.js [hunk 6]
```diff
@@ -18,6 +53,58 @@ app.get('/hello/:name', (req, res) => {
...
+      res.status(202).send({token: TOKEN_SECRET});
+    } else {
+      res.status(501).send('Error logging in');
```
Comment (reiden-webber) [inline index.js:100]: 500s messages are reserved for server-side errors and 400s messages are for user errors I think. (fact check me?). I'm guessing that this line after the else statement means the user logged in a username or password that doesn't exist

**Review by reiden-webber — CHANGES_REQUESTED** [review]

File: index.js (hunk 6 above)
Comment (Carbinski) [reply index.js:100]: oh lol I was just making up error codes

**Comment by Carbinski** [comment]
I think I made the changes using the in app editor.

**Pull request closed without merging**